client/sniffer/src/main/java/org/elasticsearch/client/sniff/Sniffer.java

```diff
@@ -151,4 +155,24 @@ synchronized void shutdown() {
     public static SnifferBuilder builder(RestClient restClient) {
         return new SnifferBuilder(restClient);
     }
+
+    private static class SnifferThreadFactory implements ThreadFactory {
+
+        private final AtomicInteger threadNumber = new AtomicInteger(1);
+        private final String namePrefix;
+        private final ThreadFactory originalThreadFactory;
+
+        private SnifferThreadFactory(String namePrefix) {
+            this.namePrefix = namePrefix;
+            this.originalThreadFactory = Executors.defaultThreadFactory();
```

Comment:
I think we should still wrap this in a doPriviledged block same asn the entire thread creation and set* below. It will allow us and other users to give fine granular security permissions